build.gradle

```diff
@@ -96,10 +96,8 @@ dependencies {
     compile 'com.opencsv:opencsv:3.4'
     compile 'com.google.guava:guava:18.0'
     compile 'com.github.samtools:htsjdk:'+ htsjdkVersion
-    // Using the shaded version to avoid conflicts between its protobuf dependency
-    // and that of Hadoop/Spark (either the one we reference explicitly, or the one
-    // provided by dataproc).
-    compile 'com.google.cloud:gcloud-java-nio:0.2.8:shaded'
+    // Don't use the shaded version, since it includes un-relocated netty, which causes conflicts with Spark
+    compile 'com.google.cloud:gcloud-java-nio:0.2.8'
```

Comment:
You're welcome @tomwhite ! I used the following commands:
```
$ gcloud auth application-default login
$ ./gatk-launch ExampleNioCountReads -P gs://${TEST_BUCKET}/hellbender-test-inputs/CEUTrio.HiSeq.WGS.b37.ch20.1m-2m.NA12878.bam -- --sparkRunner GCS --cluster jp-spark-v1
```
This code uses NIO from a Spark job (the scenario that requires shading). It's proof of concept for now, but the idea is to offer both gcs-via-hdfs-connector and gcs-via-NIO modes eventually, perhaps via a command-line flag.

Yes I can look into shading. In fact, my PR's already under review.

(sorry if this posts multiple times, GitHub seems to be having a bit of difficulty today).